mkdocs/docs/how-to-release.md

```diff
@@ -243,3 +252,13 @@ Make sure to create a PR to update the [GitHub issues template](https://github.c
 ## Update the integration tests
 
 Ensure to update the `PYICEBERG_VERSION` in the [Dockerfile](https://github.com/apache/iceberg-python/blob/main/dev/Dockerfile).
+
+## Create a Github Release Note
+
+First, follow the directions for Setting the tag again to create a tag for the approved release version (e.g. `0.7.0`).
+
+Then, create a new [Release Note](https://github.com/apache/iceberg-python/releases) on the iceberg-python Github repository by clicking on **Draft a New Release**.
```

Comment:
might be easier to link https://github.com/apache/iceberg-python/releases/new

Reply:
I think we also need to set the "Previous tag" before clicking "Generate release notes", `auto` seems to generate notes for all commits

Reply:
Ah, you are right! Thank you for taking a closer look :)